packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+        },
+      ],
+    };
+    const producer = getMockTenant(consumer.id);
+
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.suspended,
+      attributes: {
+        certified: [[getMockEServiceAttribute()]],
+        declared: [],
+        verified: [],
+      },
+    };
+    const eservice = getMockEService(generateId<EServiceId>(), producer.id, [
+      descriptor,
+    ]);
+
+    const agreement = {
+      ...getMockAgreement(eservice.id, consumer.id),
+      producerId: producer.id,
+      descriptorId: eservice.descriptors[0].id,
+      state: agreementState.draft,
+    };
+
+    await addOneEService(eservice);
+    await addOneTenant(consumer);
+    await addOneTenant(producer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+    mockSelfCareResponse(undefined);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(userNotFound(consumer.selfcareId, authData.userId));
+  });
+
+  it("should throw an agreementSelfcareIdNotFound error when selfcare id is not found for current UserId", async () => {
+    const consumer = {
+      ...getMockTenant(),
+      selfcareId: undefined,
+      mails: [
+        {
+          id: generateId(),
+          kind: tenantMailKind.ContactEmail,
+          address: "avalidemailaddressfortenant@testingagreement.com",
+          createdAt: new Date(),
+        },
+      ],
+    };
+    const producer = getMockTenant(consumer.id);
+
+    const descriptor = {
+      ...getMockDescriptor(),
+      state: descriptorState.suspended,
+      attributes: {
+        certified: [[getMockEServiceAttribute()]],
+        declared: [],
+        verified: [],
+      },
+    };
+    const eservice = getMockEService(generateId<EServiceId>(), producer.id, [
+      descriptor,
+    ]);
+
+    const agreement = {
+      ...getMockAgreement(eservice.id, consumer.id),
+      producerId: producer.id,
+      descriptorId: eservice.descriptors[0].id,
+      state: agreementState.draft,
+    };
+
+    await addOneEService(eservice);
+    await addOneTenant(consumer);
+    await addOneTenant(producer);
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(agreementSelfcareIdNotFound(consumer.id));
+  });
+
+  it("should throw an agreementMissingUserInfo error when who property is missing", async () => {
```

Comment:
`agreementMissingUserInfo` is triggered when the selfcare retrieved user misses the `.name`, `.surname`, or `.taxCode` properties

```suggestion
  it("should throw an agreementMissingUserInfo error when name, surname, or taxcode properties are missing in selfcare user response", async () => {
```